prow/cmd/pipeline/controller.go

```diff
@@ -352,6 +378,61 @@ func (c *controller) pipelineID(pj prowjobv1.ProwJob) (string, string, error) {
 	return id, url, nil
 }
 
+func createProwJobIdentifier(pj *prowjobv1.ProwJob) string {
+	var additionalIdentifier string
+	if pj.Spec.Refs != nil {
+		var pulls []int
+		for _, pull := range pj.Spec.Refs.Pulls {
+			pulls = append(pulls, pull.Number)
+		}
+		sort.Ints(pulls)
+		additionalIdentifier = fmt.Sprintf("%s/%s@%s %v", pj.Spec.Refs.Org, pj.Spec.Refs.Repo, pj.Spec.Refs.BaseRef, pulls)
+	}
+	return fmt.Sprintf("%s %s", pj.Spec.Job, additionalIdentifier)
+}
+
+func getFilteredProwJobs(id string, pjsToFilter []*prowjobv1.ProwJob) []*prowjobv1.ProwJob {
+	pjs := []*prowjobv1.ProwJob{}
+	for _, p := range pjsToFilter {
+		if runningState(p.Status.State) && id == createProwJobIdentifier(p) {
+			pjs = append(pjs, p)
+		}
+	}
+	sort.Slice(pjs, func(i, j int) bool {
+		return pjs[i].Status.StartTime.Before(&pjs[j].Status.StartTime)
+	})
+	return pjs
+}
+
+func abortDuplicatedProwJobs(c reconciler, pj *prowjobv1.ProwJob) (*prowjobv1.ProwJob, error) {
```

Comment:
well, I could maybe could think it better but with the prowjob interface I have in this tool, I really could not, at least at that time. It's because it marks every older job, including the one that is currently invoking the function (if it's older than any job in the future) as aborted. So it can technically modify a job that was just acquired from get if a new job appears.

Reply:
Just came to my mind that all patch operations return something and it's just another form of patch, with abort status.

Reply:
I am not entirely convinced I like it but if you can add a line of godoc about expected return value then I'm good with it

Reply:
I added godoc